packages/datastore/datastore/src/storage/adapters/indexedDB/IndexedDBStorageAdapter.ts

```diff
@@ -162,9 +162,34 @@ export class IndexedDBStorageAdapter implements StorageAdapter {
 
   public async updateById(storeName: string, input: any, id: string) {
     const store = await this.getStore(storeName);
-    const primaryKey = getPrimaryKey(this.stores, storeName); // TODO keypath could be an array
-    await this.convertToPromise<IDBValidKey>(store.put({ ...input, [primaryKey]: id }));
-    return this.convertToPromise(store.get(id));
+    const primaryKey = getPrimaryKey(this.stores, storeName);
```

Comment:
I think `getPrimaryKey` should be part of the model. Also see getPrimaryKeyFilter that I have added to model. 

I think I will refactor this after we merge.